allofplos/transformations.py

```diff
@@ -100,9 +100,8 @@ def filename_to_doi(filename):
         doi = PREFIX + article
     elif validate_filename(filename):
         doi = PREFIX + os.path.splitext(filename)[0]
-    # NOTE: A filename should never validate as a DOI, so the next elif is wrong.
-    elif validate_doi(filename):
-        doi = filename
+    else:
+        raise Exception("Invalid format for PLOS filename: {}".format(filename))
```

Comment:
could you add the same validate_filename check at the beginning of the of this functiont hat you did below for the `doi_to_url` check? that way you could remove all of the other calls to validate filename later on and simplify the logic.